requirements.txt

```diff
@@ -4,5 +4,3 @@
 # - tests/integration/targets/setup_botocore_pip
 botocore>=1.19.0
 boto3>=1.16.0
```

Comment:
not sure if it belongs in this PR or not, but I was trying out the latest versions today trying to change some EBS volumes, and I got an error trying to set throughput until I upgraded to `botocore>=1.19.27`. I was surprised to find that using the `requirements.txt` in the collection was not sufficient.

Reply:
Some modules or module features may require newer versions of the sdk than the overall collection does.  These should be documented in the module docs, and the module should check for `require_botocore_at_least`.  We don't really have a better way of handling this within Ansible to let fast-moving cloud collections support recent features without having a really aggressive SDK support cycle for the collection.

Reply:
@briantist A fair question, not really related to this PR, but I'll try to explain the what's going on:

Botocore is rapidly changing, the decision was made that as far as possible we'll support the version that was released about 1 year prior to the major release of the collection (3.0.0 => botocore 1.19.0).  This means that all modules should work cleanly with version 1.19.0 and **gracefully** fail when a feature can't be used because of botocore versions.  A net-new feature that depends on a newer version of botocore is permitted to be added, but it must mention in the documentation for that feature the version of botocore that's needed.  Our integration tests explicitly run using the 'supported' version, and testing a feature that needs a new version of botocore necessitates explicily using the newer version of botocore for the relevant tasks.

Setting this general "minimum supported version" gives consumers a reasonable expectation that **existing** playbooks shouldn't fail during a major relase cycle, while still enabling the addition of new features.  Since net-new features will generally mean that the consumer needs to read the docs, hopefully this was a reasonable balance for us to set.

If you think that a better explanation of this is needed in our documentation, then an Issue/PR is certainly welcome.


The requirement for botocore 1.19.27 specifically to change throughput is documented in the module docs:
https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/ec2_vol.py#L104

Reply:
@jillr @tremble thank you for the detailed explanation, that makes perfect sense. I don't necessarily feel the documentation is lacking, I just didn't check it in this case; it was related to the work that led to #627 so I hit it after removing that roadblock. My immediate reaction was to just `pip install -r` the collection's `requirements.txt` and I was surprised it didn't work, but I totally understand the position taken; without a per-module way to handle requirements in Ansible, I think you all have figured out the best possible way to address the issue at hand. Thanks again!